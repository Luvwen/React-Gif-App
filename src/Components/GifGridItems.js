import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItems = ({ title, url }) => {

  return (
    <div className={'card animate__animated animate__fadeIn'}>
      <img src={url} alt={title}></img>
      <p>{title}</p>
    </div>
  );
};

GifGridItems.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
