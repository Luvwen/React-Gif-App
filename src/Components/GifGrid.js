import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItems } from './GifGridItems';

export const GifGrid = ({ category }) => {
    const { data: images, loading } = useFetchGifs(category);

    return (
        <>
            <h3 className="category animate__animated animate__fadeIn">{category}</h3>
            <div className={'card-grid'}>
                {loading && <p className="animate__animated animate__flash">Loading</p>}
                {images.map(img => (
                    <GifGridItems key={img.id} {...img} />
                ))}
            </div>
        </>
    );
};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
};
