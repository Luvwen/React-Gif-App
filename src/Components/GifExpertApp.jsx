import React, { useState } from 'react';
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

export const GifExpertApp = ({ defaultCategories=[] }) => {
  const [categories, setCategories] = useState(
    defaultCategories
  );

  return (
    <div className="container-card">
      <h2>Gif Expert App</h2>
      <AddCategory setCategories={setCategories} />
      <ol>
        {categories.map((category) => <GifGrid 
        key={ category }
        category={ category }/>)}
      </ol>
    </div>
  );
};
