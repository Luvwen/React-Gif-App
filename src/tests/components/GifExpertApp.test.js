import React from 'react';
import { GifExpertApp } from '../../Components/GifExpertApp';
const { shallow } = require('enzyme');

describe('Pruebas en GifExpertApp', () => {
    test('Comprobar si se muestra correctamente ', () => {
        const wrapper = shallow(<GifExpertApp />);

        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrar una lista de categorias', () => {
        const categories = ['Sasuke', 'Vegeta'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
});
