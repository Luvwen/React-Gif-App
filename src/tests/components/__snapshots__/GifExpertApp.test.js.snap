// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp Comprobar si se muestra correctamente  1`] = `
<div
  className="container-card"
>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</div>
`;

exports[`Pruebas en GifExpertApp Debe mostrar una lista de categorias 1`] = `
<div
  className="container-card"
>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="Sasuke"
      key="Sasuke"
    />
    <GifGrid
      category="Vegeta"
      key="Vegeta"
    />
  </ol>
</div>
`;
